Source/Core/OrientedBoundingBox.js

```diff
@@ -0,0 +1,332 @@
+     * @returns {OrientedBoundingBox} The modified result parameter or a new OrientedBoundingBox instance if none was provided.
+     *
+     * @exception {DeveloperError} rectangle.width must be between 0 and pi.
+     * @exception {DeveloperError} rectangle.height must be between 0 and pi.
+     */
+    OrientedBoundingBox.fromEllipsoidRectangle = function(ellipsoid, rectangle, minimumHeight, maximumHeight, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(ellipsoid)) {
+            throw new DeveloperError('ellipsoid is required');
+        }
+        if (!defined(rectangle)) {
+            throw new DeveloperError('rectangle is required');
+        }
+        if (rectangle.width < 0.0 || rectangle.width > CesiumMath.PI) {
+            throw new DeveloperError('Rectangle width must be between 0 and pi');
+        }
+        if (rectangle.height < 0.0 || rectangle.height > CesiumMath.PI) {
+            throw new DeveloperError('Rectangle height must be between 0 and pi');
+        }
+        //>>includeEnd('debug');
+
+        minimumHeight = defaultValue(minimumHeight, 0.0);
+        maximumHeight = defaultValue(maximumHeight, 0.0);
+
+        // If the rectangle does not span the equator, then the bounding box will be aligned with the tangent plane at the center of the rectangle.
+        // If the rectangle does span the equator, then the bounding box will be aligned with the tangent plane to the equator at the longitudinal center of the rectangle.
+        var tangentPointCartographic = Rectangle.center(rectangle, scratchRectangleCenterCartographic);
+        if (rectangle.south < 0.0 && rectangle.north > 0.0) {
+            // The rectangle spans the equator
+            tangentPointCartographic.latitude = 0.0;
+        }
+        var tangentPoint = ellipsoid.cartographicToCartesian(tangentPointCartographic, scratchRectangleCenter);
+        var tangentPlane = new EllipsoidTangentPlane(tangentPoint, ellipsoid);
+        var plane = tangentPlane.plane;
+
+        // Corner arrangement:
+        //          N/+y
+        //      [0] [1] [2]
+        // W/-x [7]     [3] E/+x
+        //      [6] [5] [4]
+        //          S/-y
+        var perimeterNW = perimeterCartographicScratch[0];
+        var perimeterNC = perimeterCartographicScratch[1];
+        var perimeterNE = perimeterCartographicScratch[2];
+        var perimeterCE = perimeterCartographicScratch[3];
+        var perimeterSE = perimeterCartographicScratch[4];
+        var perimeterSC = perimeterCartographicScratch[5];
+        var perimeterSW = perimeterCartographicScratch[6];
+        var perimeterCW = perimeterCartographicScratch[7];
+
+        var lonCenter = tangentPointCartographic.longitude;
+        var latCenter = tangentPointCartographic.latitude;
+        perimeterSW.latitude = perimeterSC.latitude = perimeterSE.latitude = rectangle.south;
+        perimeterCW.latitude = perimeterCE.latitude = latCenter;
```

Comment:
Ok, sorry, I had missed that you're using the equator to define the tangent plane when the rectangle crosses it.  Though, I don't understand why you're doing that.  Wouldn't you get a "better" OBB if you used the center of the rectangle instead of the equator for a rectangle that goes from, say, -4 degrees latitude to 80 degrees latitude?

And speaking of center, averaging the latitudes and longitudes of the sides doesn't give you a very good center for the rectangle (it's skewed too heavily toward the poles).  You might consider using the midpoint latitude code I just added to `rectangleCameraPosition3D` in `Camera`.  It finds the midpoint along the geodesic, which I think would be better here, too.